src/query/expression/src/type_check.rs

```diff
@@ -745,3 +744,72 @@ pub const ALL_SIMPLE_CAST_FUNCTIONS: &[&str] = &[
 pub fn is_simple_cast_function(name: &str) -> bool {
     ALL_SIMPLE_CAST_FUNCTIONS.contains(&name)
 }
+
+/// notes:
+/// - this is NOT can_auto_cast_to.
+/// - this should be consistent with run_cast, partially ensured by test: test_can_cast_to .
+pub fn can_cast_to(src_type: &DataType, dest_type: &DataType) -> bool {
```

Comment:
looks duplicate with `can_auto_cast_to` function, may be we can add a `AutoCastRules` for parquet load case?

Reply:
What is the difference between `can_auto_cast_to` and `can_cast_to`? We should write it down in the comment so that we know how to edit this function later.

Reply:
@andylokandy  done, please review again cc @b41sh